tests/integration/test_core.py

```diff
@@ -89,9 +91,64 @@ async def test_mysql_relation(
     await ops_test.model.deploy("charmed-osm-mariadb-k8s", application_name="mariadb")
     await ops_test.model.add_relation("wordpress", "mariadb:mysql")
     await ops_test.model.wait_for_idle()
+
     assert (
             await get_app_status("wordpress") == ops.model.ActiveStatus.name
     ), (
         "application status should be active once correct database connection info "
         "being provided via relation"
     )
+
+
+@pytest.mark.asyncio
+async def test_get_initial_password_action(default_admin_password):
+    """
+    arrange: after WordPress charm has been deployed
+    act: run get-initial-password action
+    assert: get-initial-password action should return default admin password
+    """
+    assert (
+            len(default_admin_password) > 8
+    ), "get-initial-password action should return the default password"
+
+
+@pytest.mark.asyncio
+async def test_wordpress_functionality(
```

Comment:
Do you think it is worth making this test name more descriptive, e.g., include login, post, comment?

Reply:
Currently this test case only includes login, post, and comment, but I am planning to expand that with more functionality test like create user, non admin user, upload image, post with image etc. A more detailed name would be too long in that case.

Reply:
I think I will move this WordPress test procedure to a class method in `WordpressClient` because I believe this test procedure will run in a lot of different test cases.